### ISSUE TITLE
SCF Calculation Crash for Oxygen

### ISSUE THREAD

**Issue by lanshuyue** [deepmodeling/abacus-develop#5256]
### Describe the bug

The software unexpectedly crashes when performing a SCF calculation of oxygen crystal system.
![Image](https://github.com/user-attachments/assets/caa6513c-c352-4a5b-a4c9-8d5e2683f223)

### To Reproduce

STRU:(mp_12957)
![Image](https://github.com/user-attachments/assets/41e14161-f88e-48d3-8a48-81af97d3bc15)

```
ATOMIC_SPECIES
O   15.999        O_ONCV_PBE-1.0.upf

NUMERICAL_ORBITAL
O_gga_10au_100Ry_2s2p1d.orb

LATTICE_CONSTANT
1.8897261258369282 

LATTICE_VECTORS
3.3463770000      0.0000000000      0.0000000000      
-1.673188000      2.8980470000      0.0000000000      
0.0000000000      0.0000000000      11.205553000      

ATOMIC_POSITIONS
Direct

O
0.0000000000
6
0.0000000000 0.0000000000 0.0550460000 1 1 1 mag 0.0 
0.0000000000 0.0000000000 0.9449540000 1 1 1 mag 0.0 
0.6666670000 0.3333330000 0.3883790000 1 1 1 mag 0.0 
0.6666670000 0.3333330000 0.2782880000 1 1 1 mag 0.0 
0.3333330000 0.6666670000 0.7217120000 1 1 1 mag 0.0 
0.3333330000 0.6666670000 0.6116210000 1 1 1 mag 0.0
 ```

INPUT:

```
INPUT_PARAMETERS
#Parameters (1.General)
suffix                  O2
calculation             scf
ntype                   1
symmetry                1
pseudo_dir              ./

#Parameters (2.Iteration)
ecutwfc                 100
scf_nmax                100

#Parameters (3.Basis)
basis_type              lcao

#Parameters (4.Smearing)
smearing_method         mp
smearing_sigma          0.0036

#Parameters (5.Mixing)
mixing_type             broyden
mixing_beta             0.7
```

KPT:

```
K_POINTS
0
Gamma
9 9 9 0 0 0
```

### Environment

abacus3.5.3

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by WHUweiqingzhou** [comment]
@YuLiu98 could you have a look?

**Comment by YuLiu98** [comment]
Hi @lanshuyue , you can retry this calculation using abacus v3.8.0, which I tested successfully.

**Comment by YuLiu98** [comment]
[pp_orb.zip](https://github.com/user-attachments/files/17430620/pp_orb.zip)
I tested again and found a bug in symmetry:
```
nrotk = 8
 invmap[i]  = 0
 invmap[i]  = 1
 invmap[i]  = 2
 invmap[i]  = 3
 invmap[i]  = -1094795586
 invmap[i]  = -1094795586
 invmap[i]  = 6
 invmap[i]  = 7
isym = 0
invmap[isym] = 0
1 -0 -0
0 0 0
isym = 1
invmap[isym] = 1
-1 0 -0
0 0 0
isym = 2
invmap[isym] = 2
0 1 -0
0 0 0
isym = 3
invmap[isym] = 3
-0 -1 -0
0 0 0
isym = 4
invmap[isym] = -1094795586
```

It seems that some elements of `invmap` are not initialized normally.

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==2424304==ERROR: AddressSanitizer: SEGV on unknown address 0x57b02cfeb628 (pc 0x57c285259dfd bp 0x7ffd2faa3790 sp 0x7ffd2faa2f70 T0)
==2424304==The signal is caused by a READ memory access.
    #0 0x57c285259dfd in operator() /home/liuyu/github/abacus-develop/source/module_cell/module_symmetry/symmetry.cpp:1579
    #1 0x57c285259dfd in ModuleSymmetry::Symmetry::rhog_symmetry(std::complex<double>*, int*, int const&, int const&, int const&, int const&, int const&, int const&) /home/liuyu/github/abacus-develop/source/module_cell/module_symmetry/symmetry.cpp:1634
    #2 0x57c285a363ac in Symmetry_rho::psymmg(std::complex<double>*, ModulePW::PW_Basis const*, ModuleSymmetry::Symmetry&) const /home/liuyu/github/abacus-develop/source/module_elecstate/module_charge/symmetry_rhog.cpp:48
    #3 0x57c285a3286a in Symmetry_rho::begin(int const&, Charge const&, ModulePW::PW_Basis const*, ModuleSymmetry::Symmetry&) const /home/liuyu/github/abacus-develop/source/module_elecstate/module_charge/symmetry_rho.cpp:33
    #4 0x57c285dc58d9 in ModuleESolver::ESolver_KS_LCAO<std::complex<double>, double>::before_scf(int) /home/liuyu/github/abacus-develop/source/module_esolver/lcao_before_scf.cpp:341
    #5 0x57c285c9f34a in ModuleESolver::ESolver_KS<std::complex<double>, base_device::DEVICE_CPU>::runner(int, UnitCell&) /home/liuyu/github/abacus-develop/source/module_esolver/esolver_ks.cpp:416
    #6 0x57c285824866 in Relax_Driver::relax_driver(ModuleESolver::ESolver*) /home/liuyu/github/abacus-develop/source/module_relax/relax_driver.cpp:51
    #7 0x57c28585c44c in Driver::driver_run() /home/liuyu/github/abacus-develop/source/driver_run.cpp:68
    #8 0x57c285855dd0 in Driver::atomic_world() /home/liuyu/github/abacus-develop/source/driver.cpp:186
    #9 0x57c28585a738 in Driver::init() /home/liuyu/github/abacus-develop/source/driver.cpp:40
    #10 0x57c284f9a929 in main /home/liuyu/github/abacus-develop/source/main.cpp:42
    #11 0x733aefc29d8f in __libc_start_call_main ../sysdeps/nptl/libc_start_call_main.h:58
    #12 0x733aefc29e3f in __libc_start_main_impl ../csu/libc-start.c:392
    #13 0x57c284fce2d4 in _start (/home/liuyu/github/abacus-develop/build/abacus+0x2ba2d4)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/liuyu/github/abacus-develop/source/module_cell/module_symmetry/symmetry.cpp:1579 in operator()
==2424304==ABORTING
```

**Comment by maki49** [comment]
@lanshuyue you can set `symmetry_prec 1e-5` in INPUT before the fix in #5303 is merged. 

The default `1e-6` is too small for your structure.
```
0.6666670000 0.3333330000 0.2782880000 1 1 1 mag 0.0 
0.3333330000 0.6666670000 0.7217120000 1 1 1 mag 0.0 
```

**Comment by lanshuyue** [comment]
Thank you!!!!!!!!!!! It's running successfully now!!!🫡🥰🤗
